### PR TITLE
assignment4 1st commit

### PR DESCRIPTION
**Pull request by spsparmar**

### CODE REVIEW

File: assignment4.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+	if(index==fileArray.length) return;
+
+	fs.readFile(fileArray[index], 'utf8', function(error,data)	{  
+		console.log(data);
+		console.log("============");
+		myreadfile(index+1);
+		
+	});
+}
+readfile(0);
```
Comment (devanshutechracers) [inline assignment4.js]: there is no readFile function.

**Review by devanshutechracers — CHANGES_REQUESTED** [review]
Runtime Error, 'No Such Function Exists'. :(

**Review by devanshutechracers — APPROVED** [review]

File: assignment4.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+var fs = require('fs');
+var fileArray= ['file1','file2','file3','file4','file5'];
+
+console.log("============");	
+
+function myreadfile(index)	{
```
Comment (rockon1985) [inline assignment4.js]: camelCase the function name

File: assignment4.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+	if(index==fileArray.length) return;
```
Comment (rockon1985) [inline assignment4.js]: Indentation Again!?

File: assignment4.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+	fs.readFile(fileArray[index], 'utf8', function(error,data)	{  
+		console.log(data);
+		console.log("============");
+		myreadfile(index+1);
+
```
Comment (rockon1985) [inline assignment4.js]: remove this extra new line.

File: assignment4.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+var fs = require('fs');
+var fileArray= ['file1','file2','file3','file4','file5'];
+
+console.log("============");	
+
+function myreadfile(index)	{
+	if(index==fileArray.length) return;
+
+	fs.readFile(fileArray[index], 'utf8', function(error,data)	{
```
Comment (rockon1985) [inline assignment4.js]: extra spaces/tab between function name and it's scope beginning.

File: assignment4.js (hunk 3 above)
Comment (rockon1985) [inline assignment4.js]: there should be a space before and after operator in JS code. `a + b` but not `a+b`

File: assignment4.js [hunk 6]
```diff
@@ -0,0 +1,16 @@
+var fs = require('fs');
+var fileArray= ['file1','file2','file3','file4','file5'];
```
Comment (rockon1985) [inline assignment4.js]: - there should be a space before and after operator in JS code.`a + b` but not `a+b`
- there should be a space after comma after each array element.

**Review by rockon1985 — CHANGES_REQUESTED** [review]
Make the mentioned code improvements

**Review by rockon1985 — APPROVED** [review]

File: assignment4.js [hunk 7]
```diff
@@ -0,0 +1,12 @@
+var fs = require('fs');
+var fileArray= ['file1', 'file2', 'file3', 'file4', 'file5'];
+console.log("============");	
+function myReadfile(index)  {
```
Comment (devanshutechracers) [inline assignment4.js]: New lines required to improve readability.

File: assignment4.js [hunk 8]
```diff
@@ -0,0 +1,12 @@
+var fs = require('fs');
+var fileArray= ['file1', 'file2', 'file3', 'file4', 'file5'];
+console.log("============");
```
Comment (devanshutechracers) [inline assignment4.js]: New lines required to improve readability.

Comment (devanshutechracers) [inline assignment4.js]: New lines required above and below console to improve readability.

**Review by devanshutechracers — CHANGES_REQUESTED** [review]
Readability should be improved,

**Review by devanshutechracers — APPROVED** [review]

File: assignment4.js [hunk 9]
```diff
@@ -0,0 +1,17 @@
+var fs = require('fs');
+
+var fileArray= ['file1', 'file2', 'file3', 'file4', 'file5'];
+
+console.log("============");
+
+function myReadfile(index) {
+  if(index == fileArray.length)
```
Comment (devanshutechracers) [inline assignment4.js:8]: should use === instead of ==

File: assignment4.js [hunk 10]
```diff
@@ -0,0 +1,17 @@
...
+    return;
+  fs.readFile(fileArray[index], 'utf8', function(error, data) {
+    console.log(data);
```
Comment (devanshutechracers) [inline assignment4.js:11]: What if there is an error occurred while Reading the file.

File: assignment4.js [hunk 11]
```diff
@@ -0,0 +1,17 @@
+var fs = require('fs');
+
+var fileArray= ['file1', 'file2', 'file3', 'file4', 'file5'];
```
Comment (devanshutechracers) [inline assignment4.js:3]: space around `=` operator is required.

**Pull request merged**